src/sqlparser/src/ast/query.rs

```diff
@@ -311,8 +307,9 @@ pub enum TableFactor {
     },
     /// `<expr>[ AS <alias> ]`
     TableFunction {
-        expr: Expr,
+        name: ObjectName,
```

Comment:
The old syntax is ANSI `select * from TABLE(SOME_FUNCTION(some_arg))`, and it's parsing is already removed previously.